### PR TITLE
12.0 mig pos margin

### PR DESCRIPTION
**Pull request by legalsylvain**
migrate pos_margin

- from 11.0 to 12.0
- [REF] OCA convention (readme folder, etc...)
- [IMP] handle uom_id, based on the work of @wpichler here https://github.com/OCA/pos/pull/360
- [FIX] reuse algorithm of ``sale_margin``, so handle correctly multicurrency context.
- [ADD] test, improving coverage.

supersed : #360

### CODE REVIEW

**Comment by wpichler** [comment]
@legalsylvain Great work - tested it currently - does work as expected. @pedrobaeza Can this PR get merged ?

**Comment by pedrobaeza** [comment]
@wpichler please use GitHub approval system if you have reviewed. We need a second reviewer for merging it.

**Review by wpichler — APPROVED** [review]
Works as expected

File: pos_margin/models/pos_order_line.py [hunk 1]
```diff
@@ -22,11 +22,21 @@ class PosOrderLine(models.Model):
...
     def _compute_multi_margin(self):
-        for line in self:
-            if not line.product_id:
-                line.purchase_price = 0
-                line.margin = 0
-            else:
-                line.purchase_price = line.product_id.standard_price
-                line.margin = line.price_subtotal - (
-                    line.product_id.standard_price * line.qty)
+        for line in self.filtered(lambda x: x.product_id):
```
Comment (rousseldenis) [inline pos_margin/models/pos_order_line.py]: You can use : self.filtered('product_id') directly

**Review by rousseldenis — APPROVED** [review]
LGTM

File: pos_margin/models/pos_order_line.py (hunk 1 above)
Comment (legalsylvain) [reply pos_margin/models/pos_order_line.py]: thanks, I didn't know. 
Done. mark as resolved.

**Pull request merged**